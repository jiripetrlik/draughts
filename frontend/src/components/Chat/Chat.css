#chat {
    position: absolute;
    top: 10px;
    left: 550px;
    width: 250px;
    padding: 20px;
    border: 1px solid;
}

#chat-messages {
    text-align: left;
}
