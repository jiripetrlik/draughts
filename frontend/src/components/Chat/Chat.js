import React from 'react';
import TextField from '@material-ui/core/TextField';
import Typography from '@material-ui/core/Typography';

function Chat(props) {
    let messages = props.messages.map((message, index) =>
        <Typography key={index}>{message.name}: {message.text}</Typography>)

    return (
        <div id="chat">
            <div>
                {messages}
            </div>
            <TextField label="Message"/>
        </div>
    )
}

export default Chat
