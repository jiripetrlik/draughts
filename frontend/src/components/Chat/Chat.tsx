import React from 'react';
import TextField from '@material-ui/core/TextField';
import Typography from '@material-ui/core/Typography';
import { sendMessage } from '../../api/index'

class Chat extends React.Component<Readonly<any>, Readonly<any>> {

    keyPress(e: any) {
        if(e.keyCode === 13){
            let command = {
                Name: "message",
                Parameters: e.target.value
            }
            e.target.value = ""

            sendMessage(JSON.stringify(command))
         }
    }

    render() {
        let messages = this.props.messages.map((message: any, index: any) =>
            <Typography key={index}>{message.Name}: {message.Text}</Typography>)

        return (
            <div id="chat">
                <div>
                    {messages}
                </div>
                <TextField label="Message" onKeyDown={this.keyPress}/>
            </div>
        )
    }
}

export default Chat
