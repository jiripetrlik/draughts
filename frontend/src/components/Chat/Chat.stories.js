import React from 'react'
import Chat from './Chat'

export default {
  component: Chat,
  title: 'Chat',
};

const Template = args => <Chat {...args} />;

export const Default = Template.bind({});
Default.args = {
  messages: []
};

export const WithMessages = Template.bind({});
WithMessages.args = {
  messages: [
        {
            Name: "You",
            Text: "Message 1"
        },
        {
            Name: "Oponent",
            Text: "Longer message"
        },
        {
            Name: "You",
            Text: "One more message"
        }
    ]
};
