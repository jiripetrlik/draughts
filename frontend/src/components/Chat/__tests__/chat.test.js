import React from 'react';
import renderer from 'react-test-renderer';
import Chat from '../Chat'

test('snapshot test', () => {
    const messages = [
        {
            Name: "You",
            Text: "Message 1"
        },
        {
            Name: "Oponent",
            Text: "Longer message"
        },
        {
            Name: "You",
            Text: "One more message"
        }
    ];

    const tree = renderer
        .create(<Chat messages={messages} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
