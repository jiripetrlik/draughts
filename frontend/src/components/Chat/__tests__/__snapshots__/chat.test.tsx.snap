// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  id="chat"
>
  <div>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      You
      : 
      Message 1
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      Oponent
      : 
      Longer message
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      You
      : 
      One more message
    </p>
  </div>
  <div
    className="MuiFormControl-root MuiTextField-root"
    onKeyDown={[Function]}
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
    >
      Message
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
  </div>
</div>
`;
